src/System.CommandLine/Binding/TypeExtensions.cs

```diff
@@ -15,25 +15,19 @@ internal static class TypeExtensions
                 return type.GetElementType();
             }
 
-            Type enumerableInterface;
-
-            if (type.IsEnumerable())
-            {
-                enumerableInterface = type;
-            }
-            else
+            if (type == typeof(string))
             {
-                enumerableInterface = type
```

Comment:
Typically this kind of pattern is trimming safe, but the trimmer doesn't know that you are only looking for one interface. Since you are hard-coding looking for `typeof(IEnumerable)`, the trimmer will never trim the IEnumerable interface. It could trim other interfaces (which is why GetInterfaces is marked as unsafe). But you don't care about the other interfaces, so you can suppress this.

See the following for the same examples, and reasoning:

[https://github.com/dotnet/runtime/blob/cf5b231fcbea483df3b081939b422adfb6fd486a/src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs#L140-L157](https://github.com/dotnet/runtime/blob/cf5b231fcbea483df3b081939b422adfb6fd486a/src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs#L140-L157)

[https://github.com/dotnet/runtime/blob/cf5b231fcbea483df3b081939b422adfb6fd486a/src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs#L204-L213](https://github.com/dotnet/runtime/blob/cf5b231fcbea483df3b081939b422adfb6fd486a/src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs#L204-L213)